src/js/actions/sampler.js

```diff
@@ -175,5 +143,142 @@ define(function (require, exports) {
     };
 
+    /**
+     * Process a single click from the sampler tool.
+     * Based on the clicked leaf layer type, will sample the property from it
+     * 
+     * @private
+     * @param {Document} doc Document model
+     * @param {number} x Offset from the left window edge
+     * @param {number} y Offset from the top window edge
+     * @return {Promise}
+     */
+    var click = function (doc, x, y, secondary) {
+        var uiStore = this.flux.store("ui"),
+            coords = uiStore.transformWindowToCanvas(x, y),
+            layerTree = doc.layers;
+        
+        return uiUtil.hitTestLayers(doc.id, coords.x, coords.y)
+            .bind(this)
+            .then(function (hitLayerIDs) {
+                var hitLayerMap = new Set(hitLayerIDs.toJS()),
+                    clickedLeafLayers = layerTree.leaves.filter(function (layer) {
+                        return hitLayerMap.has(layer.id);
+                    }),
+                    topLayer = clickedLeafLayers.last();
+
+                if (secondary) {
+                    return _sampleLayerSecondary.call(this, doc, topLayer);
+                } else {
+                    return _sampleLayerPrimary.call(this, doc, topLayer, x, y);
+                }
+            });
+    };
+    click.reads = [locks.PS_DOC, locks.JS_DOC, locks.PS_APP];
+    click.writes = [locks.PS_DOC, locks.JS_DOC];
+    
+    /**
+     * Saves the currently selected layer's style in the style store clipboard
+     *
+     * @param {Document} document
+     * @param {Layer} source Layer to copy style of
+     *
+     * @return {Promise}
+     */
+    var copyLayerStyle = function (document, source) {
+        var payload = {
+            layer: source
+        };
+
+        this.dispatch(events.style.COPY_STYLE, payload);
+
+        return Promise.resolve();
+    };
+    copyLayerStyle.reads = [locks.JS_DOC];
+    copyLayerStyle.writes = [locks.JS_STYLE];
+
+    /** 
+     * Helper for menu item
+     */
+    var copyLayerStyleInCurrentDocument = function () {
+        var applicationStore = this.flux.store("application"),
+            currentDocument = applicationStore.getCurrentDocument(),
+            selectedLayers = currentDocument.layers.selected;
+
+        if (!currentDocument ||
+            selectedLayers.size !== 1) {
+            return Promise.resolve();
+        }
+        return this.transfer(copyLayerStyle, currentDocument, selectedLayers.first());
+    };
+    copyLayerStyleInCurrentDocument.reads = [locks.JS_DOC];
+    copyLayerStyleInCurrentDocument.writes = [locks.JS_STYLE];
+
+    /**
+     * Applies the saved layer style to the given layers
+     * Fill color applies to text color and vica versa
+     * Layer effects are sent across
+     * Stroke color is ignored by type layers
+     *
+     * @param {Document} document
+     * @param {Immutable.Iterable.<Layer>} targetLayers
+     *
+     * @return {Promise}
+     */
+    var pasteLayerStyle = function (document, targetLayers) {
```

Comment:
Probably not, but you can't do many of our actions on inactive documents... we still have the illusion of document parameter though.